app/channels/api/v1/global_race_channel.rb

```diff
@@ -0,0 +1,68 @@
+class Api::V1::GlobalRaceChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from('global_channel')
+  end
+
+  def unsubscribed
+    stop_all_streams
+  end
+
+  def create_race(data)
+    return if current_user.blank?
+    return if data['race_type'].blank?
+
+    race_type = Raceable.race_from_type(data['race_type'])
+    return if race_type.nil?
+
+    race = race_type.new
+    case race
+    when StandardRace
+      return if data['category_id'].blank?
+
+      category = Category.find_by(id: data['category_id'])
+      return if category.nil?
+
+      race.category = category
+    when BingoRace
+      return if data['game_id'].blank?
+      return if data['bingo_card'].blank?
+
+      game = Game.find_by(id: data['game_id'])
+      return if game.nil?
+
+      race.game = game
+      race.card = data['bingo_card']
+    when RandomizerRace
+      return if data['game_id'].blank?
+      return if data['seed'].blank?
+
+      game = Game.find_by(id: data['game_id'])
+      return if game.nil?
+
+      race.game = game
+      race.seed = data['seed']
+    end
+
+    race.owner = current_user
+    race.status_text = Raceable::OPEN
+    race.visibility!(data['visibility'])
+    race.notes = data['notes'] if data['notes'].present?
+    if race.save
+      transmit({
+        type: 'race_creation_successful',
+        data: {
+          message: 'Race has been created',
+          race:    Api::V4::RaceBlueprint.render_as_hash(race),
+          path:    Rails.application.routes.url_helpers.race_path(*race.url_params)
+        }
+      })
```

Comment:
Is it possible / reasonable to extract this generic outer message for transmits to a blueprinter template as well? Having these hash literals sitting here among all this race logic makes me nervous something will accidentally get rustled, like a single character gets accidentally changed, and all race clients immediately break completely.

On a related note, I think we should have some guarantee that `type` is something we expect and not something one of us has typo'd. I don't know if that fits into blueprinter or not, but even something like a
```
MESSAGE_TYPES = %w[race_creation_successful ...].freeze
```
or whatever with a message-generator method that checks against it would help me sleep at night.

Reply:
Yeah, I definitely want a sole location that defines all the types that can be sent, and then a way to make sure that any type of message sent back is in that array/list/enum.  I'm not sure if blueprinter is the correct solution for that since I haven't looked into it for non-instances of AR, but I'm sure something out there exists for this kind of thing.

Speaking along stability lines, do you think its worth rescuing any methods in the channels that send back some sort of status in the event that something errors?  Just like a basic hash with a type of something like `fatal_server_error` so that the client knows that something horrible went wrong and that it's request just isn't taking ages?

Reply:
Yeah, if the client expects a response that's a good idea.

Reply:
Let me know what you think of the current way I handled messages, and if it works for you then I'll make the rescue's do the same thing essentially as what the normal messages have.